packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/method_channel/method_channel_dynamic_link_builder.dart

```diff
@@ -0,0 +1,62 @@
+// ignore_for_file: require_trailing_commas
+// Copyright 2021, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:firebase_dynamic_links_platform_interface/firebase_dynamic_links_platform_interface.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_dynamic_links_platform_interface/src/platform_interface/platform_interface_dynamic_link_builder.dart';
+import 'package:firebase_dynamic_links_platform_interface/src/method_channel/method_channel_firebase_dynamic_links.dart';
+
+class MethodChannelDynamicLinkBuilder extends DynamicLinkBuilderPlatform {
+  MethodChannelDynamicLinkBuilder(FirebaseDynamicLinksPlatform dynamicLinks)
+      : super(dynamicLinks);
+
+  /// Attaches generic default values to method channel arguments.
+  Map<String, dynamic> _withChannelDefaults(Map<String, dynamic> other) {
+    return {
+      'appName': dynamicLinks.app.name,
+    }..addAll(other);
+  }
+
+  @override
+  Future<ShortDynamicLink> shortenUrl(Uri url,
+      [DynamicLinkParametersOptions? options]) async {
+    final Map<String, dynamic>? reply = await MethodChannelFirebaseDynamicLinks
```

Comment:
These Method Channel calls needs to be try/catch with error hanlding to a `FirebaseException`